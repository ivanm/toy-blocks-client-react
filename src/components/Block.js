import React from "react";
import PropTypes from "prop-types";
import { Typography, makeStyles } from "@material-ui/core";
import Card from "@material-ui/core/Card";

function Block({ id, text }) {
  const classes = useStyles();
  const formatIdNumber = number => `00${number}`;
  return (
    <Card elevation={0} className={classes.blockCard}>
      <Typography className={classes.blockHeading}>
        {formatIdNumber(id)}
      </Typography>
      <Typography className={classes.blockContent}>{text}</Typography>
    </Card>
  );
}

const useStyles = makeStyles(theme => ({
  blockCard: {
    background: "#E0E0E0",
    padding: 11,
    marginBottom: 9
  },
  blockContent: {
    fontSize: theme.typography.pxToRem(15)
  },
  blockHeading: {
    fontSize: theme.typography.pxToRem(12),
    color: "#304FFE",
    letterSpacing: 2,
    fontWeight: "bold"
  }
}));

Block.propTypes = {
  id: PropTypes.string,
  text: PropTypes.string
};

export default Block;
