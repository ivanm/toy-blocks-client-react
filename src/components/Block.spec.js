import React from "react";
import { create } from "react-test-renderer";
import Block from "../components/Block";

describe("<Block />", () => {
  it("should match snapshot", () => {
    const id = "1";
    const text = "Demo Text";
    const component = create(<Block id={id} text={text} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
