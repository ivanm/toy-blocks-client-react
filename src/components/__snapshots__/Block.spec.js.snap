// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should match snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-blockCard-1 MuiPaper-elevation0 MuiPaper-rounded"
>
  <p
    className="MuiTypography-root makeStyles-blockHeading-3 MuiTypography-body1"
  >
    001
  </p>
  <p
    className="MuiTypography-root makeStyles-blockContent-2 MuiTypography-body1"
  >
    Demo Text
  </p>
</div>
`;
